api/src/main/java/org/commonjava/indy/data/StoreDataManager.java

```diff
@@ -250,4 +250,14 @@ void reload()
      * Return the set of {@link Group}s that contain the given {@link StoreKey}s either directly or indirectly.
      */
     Set<Group> getGroupsAffectedBy( Collection<StoreKey> keys );
+
+    /**
+     * Check if store is a readonly hosted repository. Return true only when store is a readonly {@link HostedRepository}
+     */
+    boolean checkHostedReadonly( ArtifactStore store );
```

Comment:
Can we make this just `isReadonly(..)`? That way, we can expand the scope of what that means to groups / remote repositories eventually, if we ever need to. This isn't a huge deal for now, but it might help us avoid supporting old / legacy method names someday.

Remote repos are read-only now, but I've been hoping to implement "deploy-through" for remote repositories, where something like a promote request might push content to the remote server. In that case, it might make sense to mark some remote repos as readonly if they don't support this sort of thing (a lot won't).